code/crates/test/src/block/transaction.rs

```diff
@@ -0,0 +1,174 @@
+use core::fmt;
+
+use bytes::Bytes;
+use malachitebft_proto::{self as proto};
+
+use crate::hash::Hash;
+
+/// Transaction
+#[derive(Clone, PartialEq, Eq, Ord, PartialOrd)]
+pub struct Transaction {
+    data: Bytes,
+    hash: Hash,
+}
+
+impl Transaction {
+    /// Create a new transaction from bytes
+    pub fn new(data: impl Into<Bytes>) -> Self {
+        let data = data.into();
+        let hash = Self::compute_hash(&data);
+        Self { data, hash }
+    }
+
+    /// Get bytes from a transaction
+    pub fn to_bytes(&self) -> Bytes {
+        self.data.clone()
+    }
+
+    /// Get bytes from a transaction
+    pub fn as_bytes(&self) -> &[u8] {
+        self.data.as_ref()
+    }
+
+    /// Size of this transaction in bytes
+    pub fn size_bytes(&self) -> usize {
+        self.data.len()
+    }
+
+    /// Hash of this transaction
+    pub fn hash(&self) -> Hash {
+        self.hash
+    }
+
+    /// Compute the hash of a transaction
+    ///
+    /// TODO: Use hash function from Context
```

Comment:
We can remove this todo, it's actually not needed since hashing only happens on the app side and the Context does not have to care about it.